crates/autopilot/src/database/order_events.rs

```diff
@@ -27,20 +28,27 @@ impl super::Postgres {
     }
 }
 
+// Temporarily hardcoded. Will migrate to a config file in case the batches
+// approach has a positive impact.
+const DEFAULT_BATCH_SIZE: usize = 500;
+
 async fn store_order_events(
     db: &super::Postgres,
     events: &[(OrderUid, OrderEventLabel)],
     timestamp: DateTime<Utc>,
 ) -> Result<()> {
     let mut ex = db.0.begin().await.context("begin transaction")?;
-    for (uid, label) in events {
-        let event = OrderEvent {
-            order_uid: ByteArray(uid.0),
-            timestamp,
-            label: *label,
-        };
+    for w in events.chunks(DEFAULT_BATCH_SIZE) {
+        let batch = w
+            .iter()
+            .map(|(uid, label)| OrderEvent {
+                order_uid: ByteArray(uid.0),
+                timestamp,
+                label: *label,
+            })
+            .collect_vec();
```

Comment:
`insert_order_events_batch()` could take an `impl IntoIterator<Item = OrderEvent>` to avoid having to allocate a huge vector here.